Problem-1_Path_sum_II.py

```diff
@@ -0,0 +1,95 @@
+# APPROACH 1 - ITERATIVE WAY
+# Time Complexity : O(n^2), n: number of nodes of the tree
+# Space Complexity : O(nh), h: height of the tree
+# Did this code successfully run on Leetcode : Yes
+# Any problem you faced while coding this : None
+#
+#
+# Your code here along with comments explaining your approach
+# 1. Do inorder traversal. As you keep adding nodes to the stack, also add the current sum found till now and also copy of array that has elements visited till now to reach this node
+# 2. When you pop, If it's a leaf node, compare the current sum with the target. If equal, add the path to the result
+
+# Definition for a binary tree node.
+# class TreeNode:
+#     def __init__(self, val=0, left=None, right=None):
+#         self.val = val
+#         self.left = left
+#         self.right = right
+class Solution:
+    def pathSum(self, root: TreeNode, sum: int) -> List[List[int]]:
+        
+        if root is None:
+            return None
+        
+        stack, curr_sum, isFirst, curr_path, result, target = [], 0, True, [], [], sum
+        
+        while root is not None or(len(stack) > 0 or isFirst):
+            isFirst = False
+            
+            while root is not None:
+                curr_sum += root.val
+                curr_path.append(root.val)
+                list_add = copy.deepcopy(curr_path)
+                stack.append((root, curr_sum, list_add))
+                root = root.left
+                
+            curr_node = stack.pop(-1)
+            root, curr_sum, curr_path = curr_node[0], curr_node[1], curr_node[2]
+            
+            if root.left is None and root.right is None and curr_sum == target:
+                result.append(curr_path)
+                
+            root = root.right
+            
+        return result
+
+
+
+
+
+
+# APPROACH 2: RECURSIVE WAY
+# Time Complexity : O(n^2), n: number of nodes of the tree
+# Space Complexity : O(nh), h: height of the tree
```

Comment:
The space complexity for both the approaches should O(n^2).
n nodes and a new copy at every node, n*n or n^2.